### DISCUSSION TITLE
Point of tags with full-text search?

### DISCUSSION THREAD

**Discussion by xeruf** [jrnl-org/jrnl#1656] [General]
Why would I bother with tags if I can just filter with `-contains`?

**Comment by micahellison** [comment]
I generally don't use tags for that reason. I'm more of a "search, don't sort" person when it comes to recording information with jrnl.

However, tags can be used really fluently with jrnl. For instance, `jrnl @todo` is a lot less typing than `jrnl -contains todo`.

Also, sometimes someone might use a tag to delineate what an entry is about. I might search for chili recipes in my cooking journal with `jrnl cooking -contains chili` and yet it may also include a comment in an entry about a stew where I used tomato paste "just like in my chili", which I wouldn't want in my search results. If I were a bit more diligent and used a `@chili` tag, I wouldn't run into that problem.

Not that I'm going to do that. I don't mind a little noise in my search results if it means less work entering data. But that's just my personal preference.

**Reply by xeruf** [reply]
> For instance, jrnl @todo is a lot less typing than jrnl -contains todo

This is the argument I expected - which can easily be fixed by an alias or by making the former a shorthand for the latter.

But overall I fully agree with that sentiment - I even wrote a script to automatically tag specific words before I realized this is not needed ^^

**Reply by micahellison** [reply]
> This is the argument I expected - which can easily be fixed by an alias or by making the former a shorthand for the latter.

This is true for lots of situations but not all, because `-contains` has no concept of individual words, unlike tag search. For instance, `jrnl @tag` will include entries with `@tag` but not necessarily `@tag2`, whereas `jrnl -contains tag` will include both `tag` and `tag2`.
